kolibri/core/assets/src/views/CoreMenu/__tests__/CoreMenuOption.spec.js

```diff
@@ -0,0 +1,165 @@
+import { render, fireEvent, screen } from '@testing-library/vue';
```

Comment:
I think that now that we are doing tests for the CoreMenu, I think it would be good to do tests for the index as well, since it has interesting behavior with keyboard navigation for example.

Reply:
I tried adding a new test suite for checking the focusability like the following:
```ts
     it('pressing tab from keyboard should focus the menuitem', async () => {
        renderComponent({ subRoutes: sampleSubRoutes });

        // Press tab to focus the menuitem
        await fireEvent.keyDown(document.body, { key: 'Tab', code: 'Tab' });

        const menuItem = screen.getByRole('menuitem');
        expect(menuItem).toHaveFocus();
      });
```
but I can't seem to get it to work. I tried a couple of different combinations:
- Replacing `document.body` with `document`, `document.activeElement`, `menuItem` etc.
- Replacing the Key object for `fireEvent` with other variations like involving keys like `keyCode`
- Pressing the Tab key multiple times
However, this does not seem to shift the focus from the default `body` to any other element. Can you please help me figure out why the same doesn't work?

Reply:
Oh for this we can use the [userEvent](https://testing-library.com/docs/user-event/v13/) package provided by testing-library! We will need to install it here. I thought that we had used it in these tests :sweat_smile:. With this you can use the `userEvent.tab` method:

```javascript
  it('pressing tab from keyboard should focus the menuitem', async () => {
    renderComponent({ subRoutes: sampleSubRoutes });

    // Press tab to focus the menuitem
    await userEvent.tab(); // skip csrfmiddlewaretoken input
    await userEvent.tab();

    const menuItem = screen.getByRole('menuitem');
    expect(menuItem).toHaveFocus();
  });
```

Note that I added an aditional userEvent.tab because for some reason we add a csrfmiddlewaretoken input [here](https://github.com/EshaanAgg/kolibri/blob/77048d3b922e0a1bd91731804ff07cb348dc28ae/packages/kolibri-tools/jest.conf/setup.js#L58). Let me check why do we have that, but for now, this is working :).

Reply:
Just checked with Richard and told me that this csrfmiddlewaretoken input is not needed anymore, so feel free to remove it :).

Reply:
Thanks @AlexVelezLl for all the help! I had stumbled upon the `userEvent` package too, but wasn't able to decipher what it's doing differently than `fireEvent`, so I was skeptical about adopting it. But using `userEvent.tab()` indeed does work! Have updated the PR with the same, and even removed the `csrfmiddlewaretoken` input.